desci-server/src/controllers/nodes/createDpid.ts

```diff
@@ -0,0 +1,159 @@
+const url = process.env.SERVER_URL;
+
+if (url.includes("localhost")) {
+  aliasRegistryAddress = contracts.localDpidAliasInfo.proxies.at(0).address;
+} else if (url.includes("dev") || url.includes("staging")) {
+  aliasRegistryAddress = contracts.devDpidAliasInfo.proxies.at(0).address;
+} else if (process.env.NODE_ENV === "production") {
+  aliasRegistryAddress = contracts.prodDpidAliasInfo.proxies.at(0).address;
+};
+
+export const createDpid = async (req: RequestWithNode, res: Response<DpidResponse>) => {
+  const owner = req.user;
+  const node = req.node;
+  const { uuid } = req.body;
+
+  const logger = parentLogger.child({
+    module: "NODE::createDpidController",
+    body: req.body,
+    uuid,
+    user: owner,
+    ceramicStream: node.ceramicStream,
+  });
+
+  if (!uuid) {
+    return res.status(400).json({ error: "UUID is required" });
+  };
+
+  if (!process.env.HOT_WALLET_KEY) {
+    logger.error("hot wallet not configured");
+    return res.status(500).json({ error: "dpid registration not available" });
+  };
+
+  if (!process.env.ETHEREUM_RPC_URL) {
+    logger.error("ethereum RPC endpoint not configured");
+    return res.status(500).json({ error: "dpid registration not available" });
+  };
+
+  try {
+    const dpid = await getOrCreateDpid(node.ceramicStream);
+    if (!node.dpidAlias) {
+      setDpidAlias(uuid, dpid);
+    };
+    return res.status(200).send({ dpid });
+  } catch (err) {
+    logger.error({ err }, "node-create-dpid-err");
+    return res.status(400).send({ error: err.message });
+  };
+};
+
+export const getOrCreateDpid = async (
+  streamId: string,
+): Promise<number> => {
+  const logger = parentLogger.child({
+    module: "NODE::mintDpid",
+    ceramicStream: streamId,
+  });
+
+  const provider = new ethers.providers.JsonRpcProvider(
+    process.env.ETHEREUM_RPC_URL
+  );
+
+  await provider.ready;
+  const wallet = new ethers.Wallet(
+    url.includes("localhost") ? GANACHE_PKEY : process.env.HOT_WALLET_KEY,
+    provider,
+  );
+
+  const dpidAliasRegistry = tc.DpidAliasRegistry__factory.connect(
+    aliasRegistryAddress,
+    wallet,
+  );
+
+  const checkDpid = await dpidAliasRegistry.find(streamId);
+  const existingDpid = ethers.BigNumber.from(checkDpid).toNumber();
+
+  if (existingDpid !== 0) {
+    logger.info(`Skipping alias creation, stream ${streamId} already bound to ${existingDpid}`);
+    return existingDpid;
+  };
+
+  const tx = await dpidAliasRegistry.mintDpid(streamId);
+  const receipt = await tx.wait();
+  const { args: [ dpidBn ] } = receipt.events[0] as DpidMintedEvent;
+  const dpid = ethers.BigNumber.from(dpidBn).toNumber();
+
+  logger.info(
+    `Created dPID alias ${dpid} for stream ${streamId}`,
+  );
+
+  return dpid;
+};
+
+/**
+ * Related, but not directly API exposed, functionality to upgrade a legacy
+ * dPID. Neither this function nor the contract can do any validation that
+ * this stream represents the history of that dPID, so this needs to be
```

Comment:
we could do a check here to make sure history matches and the dpid is valid, but i suppose not super critical

Reply:
Good input, all done :heavy_check_mark: